src/controllers/busUsersController.ts

```diff
@@ -0,0 +1,49 @@
+import { NextFunction, Request, Response } from "express";
+import BusUser from "../models/UserBus";
+import asyncWrapper from "../utils/asyncWrapper";
+import AppError from "../errors/AppError";
+
+export async function subscribe(req: Request, res: Response, next: NextFunction) {
```

Comment:
Please add the following validation

- The lineId already exists
- The user id should get it from the authenticateUser middleware, as this route and all routes should be protected
- The user should only have one bus line, so you can check if there is a record with the userId and then upate the busLineId and if userId doesn't exist create new record , the query should look something like this
    ```
    BusUser.findOneAndUpdate(
          { userId: userId },
          { busLineId: busId },
          { new: true, upsert: true, setDefaultsOnInsert: true }
        );
    ```